core/templates/domain/topic/topic-summary.model.ts

```diff
@@ -60,10 +68,20 @@ export class TopicSummary {
       public classroom: string,
       public thumbnailFilename: string,
       public thumbnailBgColor: string,
+      public degreesOfMastery: DegreesOfMastery,
+      public skillDescription: SkillIdToDescriptionMap,
+      public subtopic: Subtopic[],
       public urlFragment: string) { }
 
   static createFromBackendDict(
       topicSummaryBackendDict: TopicSummaryBackendDict): TopicSummary {
+    let subtopics = null;
```

Comment:
When there are no subtopics, should it be an empty list instead?

Reply:
Subtopic is an optional property. So it depends on the requirement whether we need it or not in the TopicSummary. Initializing it to an empty array would be misleading because if there is a subtopic in the topic, we return an empty array when there is no requirement for it. What do you think?

Reply:
Why is subtopic an optional property? All topics should include a list of subtopics (though if there are none then that means there is a list of 0 subtopics).

Reply:
Because we require the subtopic in TopicSummary only for the Learner Dashboard page. Just like the classroom property in the TopicSummary is optional as we require it only on the topic dashboard page.

Reply:
I think this is incorrect actually and will cause issues when @mridul-netizen and @EeshaArif work on typing these objects. We should have a standard definition for TopicSummary and it should contain all the fields, there shouldn't be any fields missing (otherwise, it's a different object and should be named as such). Could you please discuss this with @aks681 and make changes as needed?

Reply:
As discussed with @aks681, we will create a separate topic summary model called LearnerTopicSummary and rename this topic summary to CreatorTopicSummary. This is because the fields required in the LearnerTopicSummary will be used to display user-facing topic summary like in Learner Dashboard Page and the CreatorTopicSummary will be used to display the creator-facing topic summary like in Topic and Skill Dashboard page. Does that sound good?

Reply:
Yup that works (as long as the definition of an object is consistent throughout the frontend codebase). Thanks!

Reply:
(Side note: suggest saying "learner-facing topic summary" rather than "user-facing topic summary". The latter is too general, it's unclear what type of user you're referring to.)

Reply:
Done. PTAL!